### PR TITLE
feat: infer_schema support arg 'file_format'.

### PR DESCRIPTION
**Pull request by youngsofun**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

e.g.

```
select * from infer_schema(location => '@s3', FILE_FORMAT => 'PARQUET', pattern => '.*parquet');
```

same usage as https://github.com/datafuselabs/databend/pull/9983#discussion_r1103620897

Closes #issue

### CODE REVIEW

File: src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs [hunk 1]
```diff
@@ -57,8 +56,8 @@ const INFER_SCHEMA: &str = "infer_schema";
 
 pub struct InferSchemaTable {
     table_info: TableInfo,
-    location: String,
-    files_info: StageFilesInfo,
+    args_parsed: InferSchemaArgsParsed,
+    table_args: TableArgs,
```
Comment (Xuanwo) [inline src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs:60]: So we will defer the parse logic later?

Comment (youngsofun) [reply src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs:60]: currently, TableArgs is reconstructed later (can be refactored later), 
so I keep a copy to simplify it.

File: src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs [hunk 2]
```diff
@@ -117,12 +116,7 @@ impl Table for InferSchemaTable {
     }
 
     fn table_args(&self) -> Option<TableArgs> {
-        let mut args = HashMap::new();
```
Comment (youngsofun) [inline src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs:120]: @Xuanwo here, all table function use this for serde and reconstruct (e.g. for distributed exexutition)

Comment (Xuanwo) [reply src/query/storages/fuse/src/table_functions/infer_schema/infer_schema_table.rs:120]: Thanks, I got it.

**Review by Xuanwo — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**